x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Min.java

```diff
@@ -107,7 +109,8 @@ protected TypeResolution resolveType() {
 
     @Override
     public DataType dataType() {
-        return field().dataType();
+        DataType t = field().dataType();
+        return t == TEXT ? KEYWORD : t;
```

Comment:
Done. I normally like to accept the commit suggestion, but in this case the remaining imports would cause a build failure, so I did this locally and will push a commit for all three places soon.

Reply:
all good, it's not like i really contributed :)